include/diagnostics/ess_updater_utils.hpp

```diff
@@ -0,0 +1,70 @@
+// VolEsti (volume computation and sampling library)
+// Copyright (c) 2021 Vissarion Fisikopoulos
+// Copyright (c) 2021 Apostolos Chalkis
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#ifndef ESS_UPDATER_UTILS_HPP
+#define ESS_UPDATER_UTILS_HPP
+
+#include <boost/accumulators/accumulators.hpp>
+#include <boost/accumulators/statistics/stats.hpp>
+#include <boost/accumulators/statistics/variance.hpp>
+#include <unsupported/Eigen/FFT>
+#include <complex>
+#include <cmath>
+
+
+/**
+    Compute autocovariance estimates for every lag for the input sequence
+    using the Geyer's stable estimator given in 
+    Charles J. Geyer, Practical Markov Chain Monte Carlo, Statistical Science 1992.
+
+    * @tparam NT number type
+    * @tparam VT vector type
+    * @param samples the sequence of correlated samples
+    * @param auto_cov the autocovariance estimates
+*/
+template <typename NT, typename VT>
+void compute_autocovariance(VT const& samples, VT &auto_cov) 
```

Comment:
Is it better the file having a different name, since we compute autocovariance?

Reply:
Thanks. Done.